community/kernel/src/main/java/org/neo4j/storageengine/api/StorageProperty.java

```diff
@@ -37,26 +38,18 @@
      * @param other value to compare with for equality.
      * @return whether or not the property value is equal to the given {@code value}.
      */
-    boolean valueEquals( Object other );
+    boolean valueEquals( Value other );
 
     /**
      * @return the property value.
      * @throws PropertyNotFoundException if this property instance represented a non-existent property.
      */
-    Object value() throws PropertyNotFoundException;
+    Value value() throws PropertyNotFoundException;
```

Comment:
why do we need to keep this value? it looks like it used only in tests, would propose to have a new `value()` that does not throw an exception and do what `valueForced` do now and remove `valueForced` completely